### PR TITLE
Use NULL instead of empty string when no name is provided

### LINKED ISSUE #33: [hello-world] name[0] == '\0' vs. name == NULL
Currently the tests as well as the example is implemented as checking the first character of `name` beeing `'\0'`, which makes it impossible to produce the (weird) output `"Hello, !"`.

Even if this output is weird it is valid in many other tracks, and since we are able to in C, I do also think we should resemble a behaviour which comes closer to most other tracks, and actually allow `hello(dest, size, "")` to fill `dest` with that weird string from above while `hello(dest, size, NULL)` will insert `"World"` as name.

Using `NULL` also does come close to other tracks that use some `Maybe` or `Option` type to achieve the behaviour described.

### LINKED ISSUE #43: Should we provide the solution's header and empty source file?
Including a (more or less) completed header will help set to ensure that solutions are const-correct and help to communicate all assumptions made about arguments in the tests.

Including an empty source file will help to ensure that the user can get _something_ compiling relatively quickly.

### LINKED ISSUE #56: Hello World -- limited buffer size seems like too much for an introduction exercise
The hello world exercise tests that the implementation is careful not to write out of bounds. While this is good practice, it seems excessive for a hello world. This may be someone's first exposure to C, to unit testing, and to Exercism, so it seems that we should save buffer overflows for a later topic.

I'd like to update the hello world exercise to use a comically over-sized buffer and save this topic for a later exercise. Thoughts?

### PR DESCRIPTION
**Pull request by ryanplusplus**
Closes #33 
Closes #43 
Closes #56

### CODE REVIEW

File: exercises/hello-world/src/example.c [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#include <stdio.h>
```
Comment (dpepoon) [inline exercises/hello-world/src/example.c]: Is this required...
~~~
#include <stddef.h>
~~~

Comment (ryanplusplus) [inline exercises/hello-world/src/example.c]: That's what provides the `NULL` constant.

Comment (ryanplusplus) [inline exercises/hello-world/src/example.c]: I need to add that include to the test, though, since it uses `NULL` without explicitly including `stddef.h`.

Comment (ryanplusplus) [inline exercises/hello-world/src/example.c]: Fixed

Comment (dpepoon) [inline exercises/hello-world/src/example.c]: I was suggesting that it be added to the .c file.

Comment (ryanplusplus) [inline exercises/hello-world/src/example.c]: Oh, my bad. You're right. Added.

**Comment by ryanplusplus** [comment]
Please have another look. This now addresses #43 and #56 and should be a much gentler introduction to the track.

**Comment by patricksjackson** [comment]
That looks good to me.

File: exercises/hello-world/src/example.c [hunk 2]
```diff
@@ -2,8 +2,8 @@
 #include <stddef.h>
 #include "hello_world.h"
 
-void hello(char* buffer, unsigned int buffer_length, const char* name) {
+void hello(char *buffer, const char *name) {
     if (name == NULL) name = "World";
```
Comment (palisaide) [inline exercises/hello-world/src/example.c]: Let's make this line the first thing we change after closing #40

**Comment by palisaide** [comment]
👍

File: exercises/hello-world/src/example.c (hunk 2 above)
Comment (ryanplusplus) [inline exercises/hello-world/src/example.c]: Good deal

**Comment by ryanplusplus** [comment]
Hey @patricksjackson -- I really appreciate your effort on reviews. Are you interested in becoming a co-maintainer for this track and getting committer rights? If you're just passing through that's fine too.

**Comment by patricksjackson** [comment]
Sure. I'd been meaning to ask about that here or at one of the other tracks at some point :)

**Comment by ryanplusplus** [comment]
Thanks, I've asked @kytrinyx to add you (I can't do so myself).
